### PR TITLE
 creating Dispatcher

### PR DESCRIPTION
**Pull request by rmnaser90**

### CODE REVIEW

File: src/APIs/APIManager.ts [hunk 1]
```diff
@@ -16,6 +51,17 @@ class APIManager {
             return book
         })
     }
+
+    async AuthUser(signInForm: { email?: string; password?: string; fullName?: string; confirmPassword?: string }) {
+        // simulate db
+        // send to back-end for authentication and return user
+        const { email, password } = signInForm
+        if (email == 'rami@monday.com') {
+            return password == 'qweqwe' ? user : { err: true, msg: 'wrong password' }
```
Comment (ZivCohenMoveo) [inline src/APIs/APIManager.ts:60]: Use ===

File: src/APIs/APIManager.ts [hunk 2]
```diff
@@ -1,6 +1,41 @@
 import axios from 'axios'
 import { Book, ApiBook } from '../Types/Types'
 
+const books = [
```
Comment (ZivCohenMoveo) [inline src/APIs/APIManager.ts:4]: Add type

File: src/APIs/APIManager.ts [hunk 3]
```diff
@@ -16,6 +51,17 @@ class APIManager {
             return book
         })
     }
+
+    async AuthUser(signInForm: { email?: string; password?: string; fullName?: string; confirmPassword?: string }) {
```
Comment (ZivCohenMoveo) [inline src/APIs/APIManager.ts:55]: async authUser

File: src/StoreManager/dispatcher.ts [hunk 4]
```diff
@@ -0,0 +1,87 @@
...
+            dispatch(action)
+            return user
+        } else {
+            alert(user.msg)
+            return user
+        }
+    }
+
+    const autoLogin = async function () {
+        const user = JSON.parse(localStorage.user)
```
Comment (ZivCohenMoveo) [inline src/StoreManager/dispatcher.ts:64]: Guard is missing- if `localStorage.user` is undefined?

File: src/StoreManager/reducer.ts [hunk 5]
```diff
@@ -31,12 +32,18 @@ export const reducer = function (
         case LOGIN:
             return produce(state, (draftState: State) => {
                 draftState.isLoogedIn = true
-                localStorage.setItem('user',JSON.stringify(state.forms.login))
-                draftState.user = state.forms.login
+                localStorage.setItem('user',JSON.stringify(action.payload.user))
+                if (action.payload.user) {
+                    draftState.user = action.payload.user
+                }
```
Comment (ZivCohenMoveo) [inline src/StoreManager/reducer.ts:38]: Need to add guard before `JSON.stringify(action.payload.user)`

```suggestion
                const curUser = action?.payload?.user;
                if (curUser) {
                    localStorage.setItem('user',JSON.stringify(curUser));
                    draftState.user = curUser;
                }
```

File: src/StoreManager/store.ts [hunk 6]
```diff
@@ -1,4 +1,29 @@
...
+
+const updatedState = produce(state,((draftState:State)=>{
+    draftState.isLoogedIn=true
+}))
+
+
+
+
+
+
```
Comment (ZivCohenMoveo) [inline src/StoreManager/store.ts:26]: why all the space?

File: src/StoreManager/store.ts [hunk 7]
```diff
@@ -1,4 +1,29 @@
...
+console.log(updatedState);
```
Comment (ZivCohenMoveo) [inline src/StoreManager/store.ts:27]: Remove console logs

File: src/UI/Components/CircleFrame/CircleFrame.tsx [hunk 8]
```diff
@@ -1,9 +1,9 @@
 import React from 'react'
 import CircleFrameStyled from './CircleFrameStyled'
 type Props = {
-    radius?: number
+    radius: number
```
Comment (ZivCohenMoveo) [inline src/UI/Components/CircleFrame/CircleFrame.tsx:4]: if you use default value like here, the prop should be optional

File: src/UI/Components/CircleFrame/CircleFrame.tsx [hunk 9]
```diff
@@ -1,9 +1,9 @@
...
 }
-const CircleFrame: React.FC<Props> = ({radius}) => {
+const CircleFrame: React.FC<Props> = ({radius=256}) => {
```
Comment (ZivCohenMoveo) [inline src/UI/Components/CircleFrame/CircleFrame.tsx:6]: Add type here-
`({radius=256}: Props)`

File: src/UI/Components/CircleFrame/CircleFrameStyled.tsx [hunk 10]
```diff
@@ -5,9 +5,9 @@ type Props = {
     radius?: number
 }
 const CircleFrameStyled = styled.div<Props>`
-    width: ${({ radius }) => radius + 'px' || '256px'};
-    height: ${({ radius }) => radius + 'px' || '260px'};
-    border-radius: ${({ radius }) => radius? radius / 2 + 'px' : '130px'};
+    width: ${({ radius }) => radius + 'px'};
```
Comment (ZivCohenMoveo) [inline src/UI/Components/CircleFrame/CircleFrameStyled.tsx:8]: I suggest using radius as string, and then you can pass here whatever you want- by pixels or by percentages.
I addition, the name radius is not exactly the meaning here- maybe pass height or width

File: src/UI/Pages/HomePage/Homepage.tsx [hunk 11]
```diff
@@ -20,6 +20,13 @@ type Props = {
 const Homepage: React.FC<Props> = ({ viewportHeight }) => {
     const navigate = useNavigate()
     const items = useSelector((state: State) => state.books)
+    const isLoggedIn = useSelector((state: State) => state.isLoogedIn)
+    const { fullName } = useSelector((state: State) => state.user)
+
+    const getStarted = function () {
+        navigate(isLoggedIn?'function':'signup')
```
Comment (ZivCohenMoveo) [inline src/UI/Pages/HomePage/Homepage.tsx:27]: What is `'function'`?

File: src/UI/Pages/HomePage/Homepage.tsx [hunk 12]
```diff
@@ -29,9 +36,11 @@ const Homepage: React.FC<Props> = ({ viewportHeight }) => {
                         <Title fontSize="50px" className="title">
                             Netboox
                         </Title>
-                        <Title className="subtitle">All your books in one place</Title>
+                        <Title className="subtitle">
+                            {isLoggedIn ? 'Welcome back ' + fullName : 'All your books in one place '}{' '}
```
Comment (ZivCohenMoveo) [inline src/UI/Pages/HomePage/Homepage.tsx:40]: What is `{' '}`?

**Review by ZivCohenMoveo — APPROVED** [review]
First of all, you did very nice work!
Note the guards for avoiding errors of "property of undefined", and I commented a few more small comments for improvement.

Well done!

**Review by ZivCohenMoveo — CHANGES_REQUESTED** [review]
First of all, you did very nice work!
Note the guards for avoiding errors of "property of undefined", and I commented a few more small comments for improvement.

Well done!

File: src/StoreManager/dispatcher.ts (hunk 4 above)
Comment (rmnaser90) [reply src/StoreManager/dispatcher.ts:64]: `        const user = JSON.parse(localStorage.user || '')
`

would this work?

**Pull request merged**